dali/operators/reader/loader/video/video_loader_decoder.cc

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/reader/loader/video/video_loader_decoder.h"
+
+namespace dali {
+void VideoLoaderDecoder::PrepareEmpty(VideoSample &sample) {
+  sample = {};
+}
+
+void VideoLoaderDecoder::ReadSample(VideoSample &sample) {
+  auto &sample_span = sample_spans_[current_index_];
+  auto &video_file = video_files_[sample_span.video_idx_];
+
+  ++current_index_;
+  MoveToNextShard(current_index_);
+
+  sample.data_.set_type(DALI_UINT8);
+  sample.data_.Resize(
+    TensorShape<4>{sequence_len_, video_file.Width(), video_file.Height(), video_file.Channels()});
+
```

Comment:
an idea: set both type and shape in one call

Reply:
Done